filebeat/beat/spooler.go

```diff
@@ -112,6 +113,12 @@ func (s *Spooler) Stop() {
 
 // flush flushes all event and sends them to the publisher
 func (s *Spooler) flush() {
+
+	// Prevents race in flush method
+	var mutex = &sync.Mutex{}
```

Comment:
@andrewkroh Makes sense. I now made the mutex part of the object. Could you try if that solved it the problem? If not, I will modify it again.